dds/DCPS/security/AccessControlBuiltInImpl.cpp

```diff
@@ -396,14 +372,13 @@ ::CORBA::Boolean AccessControlBuiltInImpl::check_create_datawriter(
   gov_iter end = ac_iter->second.gov->access_rules().end();
 
   for (gov_iter giter = begin; giter != end; ++giter) {
-    size_t d = giter->domain_list.count(domain_id);
 
-    if (d > 0) {
+    if (giter->domain_list.count(domain_id)) {
       Governance::TopicAccessRules::iterator tr_iter;
 
       for (tr_iter = giter->topic_rules.begin(); tr_iter != giter->topic_rules.end(); ++tr_iter) {
-        if ( ::ACE::wild_match(topic_name, tr_iter->topic_expression.c_str(), true,false)) {
```

Comment:
I'm assuming this was incorrect and should have allowed character classes.

Reply:
yeah, spec says to do what POSIX fnmatch does